spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java

```diff
@@ -171,11 +173,13 @@ public ThisT tableProperty(String name) {
     }
 
     protected T parse(Function<String, T> conversion, T defaultValue) {
-      if (optionName != null) {
-        // use lower case comparison as DataSourceOptions.asMap() in Spark 2 returns a lower case map
-        String optionValue = options.get(optionName.toLowerCase(Locale.ROOT));
-        if (optionValue != null) {
-          return conversion.apply(optionValue);
+      if (!optionNames.isEmpty()) {
```

Comment:
@aokolnychyi, I updated this to accept multiple calls to `option` and prefer the first one.

Reply:
Looks correct to me.